CI: ci-integration: TestPortAllocator failed: Unexpected count for proxy port rules

## CI failure

```
INTEGRATION_TESTS=true CGO_ENABLED=0 go test -mod=vendor -vet=all -tags=osusergo  ./... -timeout 600s -json -covermode=count -coverprofile=coverage.out | tparse
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/proxy/proxyports   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestPortAllocator (0.03s)

time="2024-12-04T16:10:01.594485479Z" level=info msg="Adding new proxy port rules for listener1:16541" id=listener1 subsys=proxy
    proxyports_test.go:[116](https://github.com/cilium/cilium/actions/runs/12163527791/job/33922942368#step:13:117): 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/proxy/proxyports/proxyports_test.go:116
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestPortAllocator
```

https://github.com/cilium/cilium/actions/runs/12163527791/job/33922942368

Looks like a recent change may be the cause? @jrajahalme 

https://github.com/sypakine/cilium/commit/b355e00dd105afe811a325f5b626da5c7b18dda5

--- a/pkg/proxy/proxyports/proxyports_test.go
+++ b/pkg/proxy/proxyports/proxyports_test.go
@@ -29,6 +29,12 @@ func (p *ProxyPorts) zeroProxyPort(pp *ProxyPort) bool {
 	return pp.ProxyPort == 0
 }
 
+func (p *ProxyPorts) releaseProxyPortWithWait(name string, portReuseWait time.Duration) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+	return p.releaseProxyPort(name, portReuseWait)
+}
+
 func TestPortAllocator(t *testing.T) {
 	testRunDir := t.TempDir()
 	socketDir := envoy.GetSocketDir(testRunDir)
@@ -72,7 +78,7 @@ func TestPortAllocator(t *testing.T) {
 	require.False(t, pp.acknowledged)
 	require.Zero(t, pp.ProxyPort)
 
-	err = p.releaseProxyPort("listener1", 10*time.Millisecond)
+	err = p.releaseProxyPortWithWait("listener1", 10*time.Millisecond)
 	require.NoError(t, err)
 
 	// Proxy port is not released immediately
@@ -144,7 +150,7 @@ func TestPortAllocator(t *testing.T) {
 	require.Equal(t, port2, pp.ProxyPort)
 
 	// 2nd release decreases the count to zero
-	err = p.releaseProxyPort("listener1", time.Microsecond)
+	err = p.releaseProxyPortWithWait("listener1", time.Microsecond)
 	require.NoError(t, err)
 
 	require.Eventually(t, func() bool {
@@ -190,7 +196,7 @@ func TestPortAllocator(t *testing.T) {
 	require.Equal(t, port3, pp.rulesPort)
 
 	// Release marks the port as unallocated
-	err = p.releaseProxyPort("listener1", time.Microsecond)
+	err = p.releaseProxyPortWithWait("listener1", time.Microsecond)
 	require.NoError(t, err)
 
 	require.Eventually(t, func() bool {
@@ -280,7 +286,7 @@ func TestRestoredPort(t *testing.T) {
 
 	// Release
 	require.Nil(t, pp.releaseCancel)
-	err = p.releaseProxyPort(ppName, time.Microsecond)
+	err = p.releaseProxyPortWithWait(ppName, time.Microsecond)
 	require.NoError(t, err)
 	require.Zero(t, pp.nRedirects)
 

